Not building uws correctly?

I'm not sure if this is a bug per-se because i don't know who is building uws, but the problem is manifesting itself there and the only module requiring it is this one so here i am.

### You want to:

* [x] report a *bug*?

### Current behaviour
error on build in uws

### Expected behaviour
not to error on build

### Setup
- OS: mac osx
- browser: electron
- engine.io version: 3.1.3

### Other information (e.g. stacktraces, related issues, suggestions how to fix)
in stdout
```
dyld: lazy symbol binding failed: Symbol not found: _SSL_library_init
  Referenced from: /Users/michaelmclaughlin/Sites/specless/cloud-compiler/sync/node_modules/uws/uws_darwin_57.node
  Expected in: flat namespace

dyld: Symbol not found: _SSL_library_init
  Referenced from: /Users/michaelmclaughlin/Sites/specless/cloud-compiler/sync/node_modules/uws/uws_darwin_57.node
  Expected in: flat namespace
```

in build log

```
  CXX(target) Release/obj.target/uws/src/Extensions.o
  CXX(target) Release/obj.target/uws/src/Group.o
In file included from ../src/Group.cpp:1:
In file included from ../src/Group.h:4:
In file included from ../src/WebSocket.h:4:
In file included from ../src/WebSocketProtocol.h:5:
../src/Networking.h:7:10: fatal error: 'openssl/opensslv.h' file not found
#include <openssl/opensslv.h>
         ^~~~~~~~~~~~~~~~~~~~
1 error generated.
make: *** [Release/obj.target/uws/src/Group.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at emitTwo (events.js:125:13)
gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 16.7.0
gyp ERR! command "/usr/local/Cellar/node/8.4.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/michaelmclaughlin/Sites/specless/cloud-compiler/sync/node_modules/uws
gyp ERR! node -v v8.4.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
```

--- a/README.md
+++ b/README.md
@@ -226,7 +226,7 @@ to a single process.
         option. If false, no path will be sent, which means browsers will only send the cookie on the engine.io attached path (`/engine.io`).
         Set false to not save io cookie on all requests. (`/`)
       - `cookieHttpOnly` (`Boolean`): If `true` HttpOnly io cookie cannot be accessed by client-side APIs, such as JavaScript. (`true`) _This option has no effect if `cookie` or `cookiePath` is set to `false`._
-      - `wsEngine` (`String`): what WebSocket server implementation to use. Specified module must conform to the `ws` interface (see [ws module api docs](https://github.com/websockets/ws/blob/master/doc/ws.md)). Default value is `uws` (see [µWebSockets](https://github.com/uWebSockets/uWebSockets)).
+      - `wsEngine` (`String`): what WebSocket server implementation to use. Specified module must conform to the `ws` interface (see [ws module api docs](https://github.com/websockets/ws/blob/master/doc/ws.md)). Default value is `ws`. An alternative c++ addon is also available by installing `uws` module.
       - `initialPacket` (`Object`): an optional packet which will be concatenated to the handshake packet emitted by Engine.IO.
 - `close`
     - Closes all clients

--- a/lib/server.js
+++ b/lib/server.js
@@ -36,7 +36,7 @@ function Server (opts) {
 
   opts = opts || {};
 
-  this.wsEngine = opts.wsEngine || process.env.EIO_WS_ENGINE || 'uws';
+  this.wsEngine = opts.wsEngine || process.env.EIO_WS_ENGINE || 'ws';
   this.pingTimeout = opts.pingTimeout || 60000;
   this.pingInterval = opts.pingInterval || 25000;
   this.upgradeTimeout = opts.upgradeTimeout || 10000;
@@ -103,16 +103,10 @@ Server.prototype.init = function () {
   if (this.ws) this.ws.close();
 
   var wsModule;
-  try {
-    switch (this.wsEngine) {
-      case 'uws': wsModule = require('uws'); break;
-      case 'ws': wsModule = require('ws'); break;
-      default: throw new Error('unknown wsEngine');
-    }
-  } catch (ex) {
-    this.wsEngine = 'ws';
-    // keep require('ws') as separate expression for packers (browserify, etc)
-    wsModule = require('ws');
+  switch (this.wsEngine) {
+    case 'uws': wsModule = require('uws'); break;
+    case 'ws': wsModule = require('ws'); break;
+    default: throw new Error('unknown wsEngine');
   }
   this.ws = new wsModule.Server({
     noServer: true,

--- a/package.json
+++ b/package.json
@@ -45,14 +45,12 @@
     "expect.js": "^0.3.1",
     "mocha": "^4.0.1",
     "s": "0.1.1",
-    "superagent": "^3.8.1"
-  },
-  "optionalDependencies": {
+    "superagent": "^3.8.1",
     "uws": "~9.14.0"
   },
   "scripts": {
     "lint": "eslint lib/ test/ *.js",
-    "test": "npm run lint && mocha && EIO_WS_ENGINE=ws mocha"
+    "test": "npm run lint && mocha && EIO_WS_ENGINE=uws mocha"
   },
   "repository": {
     "type": "git",

